app/src/main/java/com/ekeitho/reduxy/Reducer.kt

```diff
@@ -1,37 +1,40 @@
 package com.ekeitho.reduxy
 
+import io.reactivex.Observable
 import io.reactivex.disposables.Disposable
 import io.reactivex.disposables.Disposables
 import io.reactivex.subjects.BehaviorSubject
-import io.reactivex.subjects.PublishSubject
+import java.util.*
 
 object Reducer {
 
     var subscription : Disposable = Disposables.disposed()
 
     // Reducers do not state state, and they do NOT mutate state.
-    fun init(eventSubject : PublishSubject<Event>, stateSubject: BehaviorSubject<ApplicationState>) {
+    fun init(eventSubject : Observable<Event>, stateSubject: BehaviorSubject<ApplicationState>) {
         var state = stateSubject.value
+        val eventLog = LinkedList<Event>()
 
         // this is for idempotency
         // so if someone were to make a mistake of calling init more than they should
         // and we are dealing with 3 threads, then they would get back the same ref to the subscription
         if (subscription.isDisposed) {
             subscription = eventSubject.subscribe({ event: Event ->
-                val temp = state
+                eventLog.push(event)
+                val temp = state.copy()
```

Comment:
Temp and state references were attached to the same object.  We should give some thought to how much we value immutability here before scaling, though.